core/textile/bucket/bucket_dir.go

```diff
@@ -80,3 +80,31 @@ func (b *Bucket) DeleteDirOrFile(ctx context.Context, path string) (path.Resolve
 
 	return b.bucketsClient.RemovePath(ctx, b.Key(), path)
 }
+
+// return the recursive items count for a path
+func (b *Bucket) ItemsCount(ctx context.Context, path string) (int32, error) {
```

Comment:
Maybe it can include a flag `recursive: bool` to either count the files in the curr dir or all the files in subdirs